### ISSUE TITLE
Deleting files breaks search (server throws exceptions)

### ISSUE THREAD

**Issue by snej** [kantord/SeaGOAT#245]
## TL;DR: 

I stopped the server during initial indexing, used `rm -r` to delete a subdirectory of my repo, restarted the server; and now whenever I search **the server throws file-not-found exceptions trying to read deleted files.**

## What I Did

1. I installed SeaGOAT on an ARM MacBook Pro (understanding that macOS is unsupported.)
2. Started the server from a shell.
3. Noticed it had a huge number (30000+) of files to index, and that it was scanning a huge directory of Doxygen-generated HTML/js that I don’t care about.
4. Hit Ctrl-C twice to stop the server.
5. `rm -r docs/`
6. Started server again and let it finish indexing.
7. In another shell in the repo directory, entered a `gt` command to search

## Results

The `gt` command hung, not printing anything.

Over in the other window, the server had thrown a Python exception:

```
2023-09-23 10:18:06,773 Handling task: query
Exception in thread Thread-1 (_worker_function):
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.11/3.11.5/Frameworks/Python.framework/Versions/3.11/lib/python3.11/threading.py", line 1038, in _bootstrap_inner
    self.run()
  File "/opt/homebrew/Cellar/python@3.11/3.11.5/Frameworks/Python.framework/Versions/3.11/lib/python3.11/threading.py", line 975, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/snej/.local/pipx/venvs/seagoat/lib/python3.11/site-packages/seagoat/queue/base_queue.py", line 79, in _worker_function
    self._handle_task(context, task)
  File "/Users/snej/.local/pipx/venvs/seagoat/lib/python3.11/site-packages/seagoat/queue/base_queue.py", line 66, in _handle_task
    result = handler(context, *task.args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/snej/.local/pipx/venvs/seagoat/lib/python3.11/site-packages/seagoat/queue/task_queue.py", line 78, in handle_query
    context["seagoat_engine"].fetch_sync(
  File "/Users/snej/.local/pipx/venvs/seagoat/lib/python3.11/site-packages/seagoat/engine.py", line 163, in fetch_sync
    loop.run_until_complete(self.fetch(*args, **kwargs))
  File "/Users/snej/.local/pipx/venvs/seagoat/lib/python3.11/site-packages/nest_asyncio.py", line 99, in run_until_complete
    return f.result()
           ^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.5/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/futures.py", line 203, in result
    raise self._exception.with_traceback(self._exception_tb)
  File "/opt/homebrew/Cellar/python@3.11/3.11.5/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 267, in __step
    result = coro.send(None)
             ^^^^^^^^^^^^^^^
  File "/Users/snej/.local/pipx/venvs/seagoat/lib/python3.11/site-packages/seagoat/engine.py", line 147, in fetch
    self._results.extend(source["fetch"](self.query_string, limit_clue))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/snej/.local/pipx/venvs/seagoat/lib/python3.11/site-packages/seagoat/sources/chroma.py", line 58, in fetch
    files[path] = Result(path, Path(repository.path) / path)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/snej/.local/pipx/venvs/seagoat/lib/python3.11/site-packages/seagoat/result.py", line 75, in __init__
    self.line_texts = self._read_lines()
                      ^^^^^^^^^^^^^^^^^^
  File "/Users/snej/.local/pipx/venvs/seagoat/lib/python3.11/site-packages/seagoat/result.py", line 84, in _read_lines
    with open(self.full_path, encoding="utf-8") as source_code_file:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/Users/snej/path/to/repo/docs/C/html/search/enumvalues_0.js'
```

I can’t get the problem to go away — to get the server to notice a change I tried `mkdir docs`, then `touch docs/foo/html`, but nothing happened (didn’t see any output from the server.)

I’d wipe my state by deleting the server’s database, but I don’t know where it is...

**Comment by kantord** [comment]
ok, it seems like this is a serious bug, but should be straightforward to fix :rocket:

**Comment by kantord** [comment]
fixed in the latest release!
